### PR TITLE
add flight iterations

### PR DESCRIPTION
**Pull request by cory-caddell**
### What this PR does and why:
PR to iterate through various flights.

### Author's Checklist
- [x ] Add comments (either in the code or in Github) describing any code that might not be quickly understood just by reading

### Reviewer's Checklist
- [ ] Read the ticket to familiarize yourself with the business requirements these changes are addressing
- [ ] QA the feature or bugfix
- [ ] Identify other places in the codebase that could be affected by these changes
- [ ] Verify that unit tests are in place for these changes

### Testing Instructions
Run script.  Compare output file to website.

### CODE REVIEW

File: united-fare-finder.py [hunk 1]
```diff
@@ -36,45 +36,96 @@
...
-    for j in range(len(flight_elements[i])):
-        print(*flight_elements[i][j])
-    print()
+origin = "ORD"
+dest = [
+  "hnl", "slc", "dfw", "lax", "sea", "atl",
+  "mia", "mco", "sfo", "jfk", "ewr", "lga", "dca", "iad"
+]
+depDate = "03/01/2023"
+retDate = "03/10/2023"
```
Comment (daniel-doughty) [inline united-fare-finder.py]: its common to name constatnts in all caps (and snake cased) at the beginning of the file

File: united-fare-finder.py [hunk 2]
```diff
@@ -36,45 +36,96 @@
...
+
+count = 0
+for dest in dest:
+
+    # input flight info in main page
+    if (count == 0):
```
Comment (daniel-doughty) [inline united-fare-finder.py]: I think its cleaner to use `index` with `enumerate` here
https://stackoverflow.com/questions/522563/accessing-the-index-in-for-loops

File: united-fare-finder.py [hunk 3]
```diff
@@ -36,45 +36,96 @@
...
+          By.CSS_SELECTOR,
+          "button."
+          "app-components-BookFlightForm-bookFlightForm__findFlightBtn--1lbFe")
+        elem.submit()
+
+    # input flight info in subsequent pages
+    else:
+        elem = driver.find_element(
+          By.ID, "airport-autocomplete-.swap0.fromTrip")
+        elem.send_keys(Keys.CONTROL + "a")
```
Comment (daniel-doughty) [inline united-fare-finder.py]: if you are trying to overwrite, I think clear() is the better way to go
```suggestion
        elem.clear()
```

File: united-fare-finder.py [hunk 4]
```diff
@@ -36,45 +36,96 @@
...
+
+    time.sleep(15)
+
+    # scrape fare data
+    page_source = driver.page_source
+    soup = BeautifulSoup(page_source, "html.parser")
+    flight_elements = parsePageSource(soup)
+
+    # write to file
+    print(dest + " ********************************************************\n")
```
Comment (daniel-doughty) [inline united-fare-finder.py]: the second iteration errored out for me:
```no such element: Unable to locate element: {"method":"xpath","selector":"//li[@id="airport-autocomplete-.swap0.fromTrip-menu-item-0"]//div"}```

Generally the more elements you have in a script the more brittle it is going to be. any code update that they do to the page has the potential to break it (esp when using XPath or CSS selector, Id is the safest)

I think the least brittle way to do this is to just iterate through the same script that we had working for 1, and do it for every destination.  Completely starting over on the homepage with a reload, that way we can be sure of the beginning state of the page.

Maybe we don't login on every iteration though, because login itself is probably the most error prone part of this process.

File: united-fare-finder.py [hunk 5]
```diff
@@ -36,45 +36,96 @@
...
+    else:
+        elem = driver.find_element(
+          By.ID, "airport-autocomplete-.swap0.fromTrip")
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(origin)
+
+        elem = driver.find_element(
+          By.XPATH,
+          '//li[@id="airport-autocomplete-.swap0.fromTrip-menu-item-0"]//div')
+        elem.click()
```
Comment (daniel-doughty) [inline united-fare-finder.py]: what is this click for, I don't think it's necessary?

File: united-fare-finder.py [hunk 6]
```diff
@@ -36,45 +36,96 @@
...
+
+        elem = driver.find_element(By.ID, "airport-autocomplete-.swap0.toTrip")
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(dest)
+
+        elem = driver.find_element(
+          By.XPATH,
+          '//li[@id="airport-autocomplete-.swap0.toTrip-menu-item-0"]//div')
+        elem.click()
```
Comment (daniel-doughty) [inline united-fare-finder.py]: same with this click, do we need it?

File: united-fare-finder.py [hunk 7]
```diff
@@ -36,45 +36,96 @@
...
+    flight_elements = parsePageSource(soup)
+
+    # write to file
+    print(dest + " ********************************************************\n")
+    for i in range(len(flight_elements)):
+        for j in range(len(flight_elements[i])):
+            print(*flight_elements[i][j])
+        print()
+
+    count = 1
```
Comment (daniel-doughty) [inline united-fare-finder.py]: if you are going to keep track of a count, you should at least iterate it so its an actual count (ie `count++`)

**Comment by cory-caddell** [comment]
I actually changed this right after I initiated the pull request.  I was
really just trying to trigger the else statement in the following
iterations.  I realized it was hard to follow, so I changed the counter to
a boolean, mainPage = True

mainPage = True
for dest in dest:

# input flight info in main page
if (mainPage):
elem = driver.find_element(By.ID, "bookFlightOriginInput")
elem.clear()
elem.send_keys(origin)

elem = driver.find_element(By.ID, "bookFlightDestinationInput")
elem.clear()
elem.send_keys(dest)

elem = driver.find_element(By.ID, "DepartDate")
elem.clear()
elem.send_keys(depDate)

elem = driver.find_element(By.ID, "ReturnDate")
elem.clear()
elem.send_keys(retDate)

elem = driver.find_element(
By.XPATH, ***@***.***="bookFlightForm"]/div[1]/div[1]/label')
elem.click()

elem = driver.find_element(
By.CSS_SELECTOR,
"button."
"app-components-BookFlightForm-bookFlightForm__findFlightBtn--1lbFe")
elem.submit()

mainPage = False

**Comment by cory-caddell** [comment]
Those elem.click() lines are for the autocomplete drop down box.  The input
won't stick if they're not there.  It took me forever trying to figure it
out.

**Comment by cory-caddell** [comment]
It oddly works on my end, but makes sense.  I was trying to get too fancy.
I actually had that originally, but changed it trying to save an extra page
load.

**Comment by cory-caddell** [comment]
I was having trouble with this due to the autocomplete drop down box.  If
you use clear, it reverts back to the original input.

**Comment by cory-caddell** [comment]
okay!

**Comment by cory-caddell** [comment]
I actually changed this to a boolean.  I wasn't trying to count anything.
I was trying to trigger the main page code in the if statement.

But noted, I'll need to know this at some point.

File: united-fare-finder.py (hunk 1 above)
Comment (cory-caddell) [reply united-fare-finder.py]: done

File: united-fare-finder.py (hunk 2 above)
Comment (cory-caddell) [reply united-fare-finder.py]: I actually changed this to a boolean right after I initiated the pull request.  I wasn't trying to index/count anything.  I was trying to trigger the if statement.  :)

But noted, I'll need to know this at some point.

File: united-fare-finder.py (hunk 3 above)
Comment (cory-caddell) [reply united-fare-finder.py]: I was having trouble with this due to the autocomplete feature in the element.  If you use clear(), it reverts back to the original input. 

In the updated PR I'm about to send I actually changed to CONTROL + "a" due to the same issue.  Every now then during the iterations the input fields get all wonky.

File: united-fare-finder.py (hunk 4 above)
Comment (cory-caddell) [reply united-fare-finder.py]: noted!  I was trying to get a little fancy.  New PR request will have iteration through the original code, with a couple of tweaks.

File: united-fare-finder.py (hunk 5 above)
Comment (cory-caddell) [reply united-fare-finder.py]: This took me forever to figure out, but it's needed when there's an autocomplete/drop-down box feature in the element.  Don't need it for updated code, though.

File: united-fare-finder.py (hunk 6 above)
Comment (cory-caddell) [reply united-fare-finder.py]: see above.  We don't need any longer.

File: united-fare-finder.py (hunk 7 above)
Comment (cory-caddell) [reply united-fare-finder.py]: I actually changed this right after I initiated the pull request.  I was
really just trying to trigger the else statement in the following
iterations.  I realized it was hard to follow, so I changed the counter to
a boolean, mainPage = True
Screenshot from 2023-01-29 19-57-03

**Review by daniel-doughty — APPROVED** [review]
